nsxt/resource_nsxt_policy_transport_zone.go

```diff
@@ -0,0 +1,289 @@
+				ForceNew:     true,
+			},
+			"uplink_teaming_policy_names": {
+				Type:        schema.TypeList,
+				Description: "Names of the switching uplink teaming policies that are supported by this transport zone.",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"site_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the site this Transport Zone belongs to",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"enforcement_point_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the enforcement point this Transport Zone belongs to",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyTransportZoneExists(siteID, epID, tzID string, connector client.Connector) (bool, error) {
+	var err error
+
+	// Check site existence first
+	siteClient := infra.NewSitesClient(connector)
+	_, err = siteClient.Get(siteID)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to read site %s", siteID)
+		return false, logAPIError(msg, err)
+	}
+
+	// Check (ep, tz) existence. In case of ep not found, NSX returns BAD_REQUEST
+	tzClient := enforcement_points.NewTransportZonesClient(connector)
+	_, err = tzClient.Get(siteID, epID, tzID)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func policyTransportZonePatch(siteID, epID, tzID string, d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	tags := getPolicyTagsFromSchema(d)
+	isDefault := d.Get("is_default").(bool)
+	transportType := d.Get("transport_type").(string)
+	uplinkTeamingNames := getStringListFromSchemaList(d, "uplink_teaming_policy_names")
+
+	if len(uplinkTeamingNames) > 0 && transportType != model.PolicyTransportZone_TZ_TYPE_VLAN_BACKED {
+		// uplink_teaming_policy_names only valid for VLAN_BACKED TZ
+		return fmt.Errorf("Cannot use uplink_teaming_policy_names with transport_type %s", transportType)
+	}
+
+	obj := model.PolicyTransportZone{
+		DisplayName:              &displayName,
+		Description:              &description,
+		Tags:                     tags,
+		IsDefault:                &isDefault,
+		TzType:                   &transportType,
+		UplinkTeamingPolicyNames: uplinkTeamingNames,
+	}
+
+	// Create the resource using PATCH
+	tzClient := enforcement_points.NewTransportZonesClient(connector)
+	_, err := tzClient.Patch(siteID, epID, tzID, obj)
+	return err
+}
+
+func policyTransportZoneIDTuple(d *schema.ResourceData, m interface{}) (id, siteID, epID string, err error) {
+	id = d.Id()
+	if id == "" {
+		err = fmt.Errorf("Error obtaining PolicyTransportZone ID")
+		return
+	}
+	siteID = d.Get("site_id").(string)
+	if siteID == "" {
+		siteID = defaultSite
```

Comment:
is there a reason not to specify this default in the schema instead?

Reply:
Sure. Will make it default in schema.

Reply:
Why does this need to be computed, in case default is set whenever the user does not specify the value? Does platform ever override this value?

Reply:
Yes, I don't think enforcement point has a default